src/share/classes/com/sun/javatest/regtest/exec/ScratchDirectory.java

```diff
@@ -208,11 +208,11 @@ private boolean deleteFiles(File dir, Pattern p, boolean match, boolean deleteDi
 
         boolean ok = true;
         File[] children = dir.listFiles();
-        if (children == null) { // should always be not null, but sometimes it is
+        if (children == null) { // should always be not null, but File.listFiles() allows for it
```

Comment:
I find the proposed new comment somewhat confusing. How about:

`// should always be not null, but maybe null if an error occurs`

Reply:
Hello Jon, I've now updated the PR to use this suggested code comment.